### PR TITLE
Detect Homebrew on Mac M1 (Apple Silicon)

### LINKED ISSUE #73887: fact gathering does not auto-detect homebrew on Mac with M1
### Summary

module_utils/facts/system/pkg_mgr.py does not detect homebrew on Mac with M1.

homebrew on Mac with M1 is installed to /opt/homebrew/bin/brew.
But, module_utils/facts/system/pkg_mgr.py only search '/usr/local/bin/brew'.

### Issue Type

Bug Report

### Component Name

facts

### Ansible Version

```console (paste below)
$ ansible --version
ansible 2.9.6
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/kimata/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3/dist-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.8.5 (default, Jan 27 2021, 15:41:15) [GCC 9.3.0]
```

### Configuration

```console (paste below)
$ ansible-config dump --only-changed

```

### OS / Environment

Ubuntu 20.04.2 LTS

### Steps to Reproduce

```console (paste below)
$ ansible macbook -m setup
```

### Expected Results

"ansible_pkg_mgr": "homebrew",

### Actual Results

```console (paste below)
$ ssh macbook brew --prefix
/opt/homebrew

$ ansible macbook -m setup
(snip)
        "ansible_machine": "arm64",
        "ansible_memfree_mb": 6712,
        "ansible_memtotal_mb": 8192,
        "ansible_model": "MacBookAir10,1",
        "ansible_nodename": "macbook.green-rabbit.net",
        "ansible_os_family": "Darwin",
        "ansible_osrevision": "199506",
        "ansible_osversion": "20D91",
        "ansible_pkg_mgr": "unknown",  <---
        "ansible_processor": "Apple M1",
        "ansible_processor_cores": "8",
        "ansible_processor_vcpus": "8",
        "ansible_product_name": "MacBookAir10,1",
```

### PR DESCRIPTION
**Pull request by fernandoacorreia**
##### SUMMARY
The `ansible_pkg_mgr` fact is `unknown` on Macs with M1 chip (Apple Silicon -- ARM architecture).

The root cause is that Homebrew's default install path on ARM is `/opt/homebrew`.

Fixes #73887
Closes #73896

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
facts

##### ADDITIONAL INFORMATION

Homebrew's default install location for macOS on ARM is `/opt/homebrew`. Source: https://docs.brew.sh/FAQ.

On a Mac M1 (Apple Silicon), homebrew will be installed at `/opt/homebrew/bin/brew`.

Without this fix, the `ansible_pkg_mgr` fact will be `unknown`.

```
❯ ansible localhost -m setup | grep "ansible_pkg_mgr"
[WARNING]: No inventory was parsed, only implicit localhost is available
        "ansible_pkg_mgr": "unknown",
```

With the fix, the value will be `homebrew`:

```
❯ ansible localhost -m setup | grep "ansible_pkg_mgr"
[WARNING]: No inventory was parsed, only implicit localhost is available
        "ansible_pkg_mgr": "homebrew",
```

### CODE REVIEW

**Comment by fernandoacorreia** [comment]
@Akasurde please review.

File: test/units/module_utils/facts/test_collectors.py [hunk 1]
```diff
@@ -238,6 +238,46 @@ def test_collect(self):
...
+    valid_subsets = ['pkg_mgr']
+    fact_namespace = 'ansible_pkgmgr'
+    collector_class = PkgMgrFactCollector
+    collected_facts = {
+        "ansible_distribution": "MacOSX",
+        "ansible_distribution_major_version": "11",
+        "ansible_os_family": "Darwin"
+    }
+
+    @patch('ansible.module_utils.facts.system.pkg_mgr.os.path.exists', side_effect=lambda x: x == '/opt/homebrew/bin/brew')
```
Comment (samdoran) [inline test/units/module_utils/facts/test_collectors.py:253]: I don't think the `lambda` is necessary here. You should be able to set [`return_value`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.return_value) instead.

Comment (fernandoacorreia) [reply test/units/module_utils/facts/test_collectors.py:253]: @samdoran Thanks for reviewing. How would you set `return_value` to be `True` only for that specific path and not for the others?

Comment (samdoran) [reply test/units/module_utils/facts/test_collectors.py:253]: Oh, I misread this. I thought it was returning the path not the result of an evaluation. I tend to try and avoid `lambda` at all costs (harder to read/maintain usually), but I don't see a better way to write the test.

**Comment by samdoran** [comment]
Please create a [backport PR](https://docs.ansible.com/ansible/latest/community/development_process.html#backporting-merged-prs) against `stable-2.10` and `stable-2.11`.

File: test/units/module_utils/facts/test_collectors.py (hunk 1 above)
Comment (fernandoacorreia) [reply test/units/module_utils/facts/test_collectors.py:253]: Makes sense, thanks for checking. I thought this was the most concise way to do it.

**Comment by fernandoacorreia** [comment]
@samdoran Thanks for accepting the PR. I'll create the backports.

**Pull request merged**